src/libData/AccountData/AccountStoreSCEvm.cpp

```diff
@@ -0,0 +1,661 @@
+          // for now catch any generic exceptions and report them
+          // will examine exact possibilities and catch specific exceptions.
+          LOG_GENERAL(
+              WARNING,
+              "Exception thrown trying to update Contract code " << e.what());
+        }
+
+        // Actually Update the state for the contract
+        try {
+          for (const auto& sit : it->Storage()) {
+            if (not Contract::ContractStorage::GetContractStorage()
+                        .UpdateStateValue(
+                            Address(it->Address()),
+                            DataConversion::StringToCharArray(sit.Key()), 0,
+                            DataConversion::StringToCharArray(sit.Value()),
+                            0)) {
+              LOG_GENERAL(WARNING,
+                          "Exception thrown trying to update state in Contract "
+                          "storage "
+                              << it->Address());
+            }
+          }
+        } catch (std::exception& e) {
+          // for now catch any generic exceptions and report them
+          // will examine exact possibilities and catch specific exceptions.
+          LOG_GENERAL(WARNING,
+                      "Exception thrown trying to update state on the contract "
+                          << e.what());
+        }
+
+        try {
+          if (it->hasBalance() && it->Balance().size()) {
+            targetAccount->SetBalance(uint128_t(it->Balance()));
+          }
+        } catch (std::exception& e) {
+          // for now catch any generic exceptions and report them
+          // will examine exact possibilities and catch specific exceptions.
+          LOG_GENERAL(
+              WARNING,
+              "Exception thrown trying to update balance on target Account "
+                  << e.what());
+        }
+
+        try {
+          if (it->hasNonce() && it->Nonce().size()) {
+            targetAccount->SetNonce(std::stoull(it->Nonce(), nullptr, 0));
+          }
+        } catch (std::exception& e) {
+          // for now catch any generic exceptions and report them
+          // will examine exact possibilities and catch specific exceptions.
+          LOG_GENERAL(WARNING,
+                      "Exception thrown trying to set Nonce on target Account "
+                          << e.what());
+        }
+        // Mark the Address as updated
+        m_storageRootUpdateBufferAtomic.emplace(it->Address());
+      }
+    }
+  }
+
+  // send code to be executed
+  if (invoke_type == RUNNER_CREATE) {
+    contractAccount->SetImmutable(DataConversion::StringToCharArray(
+                                      "EVM" + evmReturnValues.ReturnedBytes()),
+                                  contractAccount->GetInitData());
+  }
+  return gas;
+}
+
+template <class MAP>
+bool AccountStoreSC<MAP>::ViewAccounts(EvmCallParameters& params, bool& ret,
+                                       std::string& result) {
+  TransactionReceipt rcpt;
```

Comment:
No transaction is involved, therefore there shouldn't be this `rcpt` variable.  If it is just needed by EvmCallRunner below, it would help to make this parameter optional (just a nullable pointer out output) and not create any recepipts when not necessary..

Having to create a suprios "receipt" and then discard it seems wrong.